test/integration/image-optimizer/test/index.test.js

```diff
@@ -658,9 +658,9 @@ function runTests({ w, isDev, domains = [], ttl, isSharp }) {
 
   it('should set cache-control to immutable for static images', async () => {
     if (!isDev) {
-      const filename = 'test'
+      const filename = 'test.480a01e5ea850d0231aec0fa94bd23a0'
       const query = {
-        url: `/_next/static/image/public/${filename}.480a01e5ea850d0231aec0fa94bd23a0.jpg`,
+        url: `/_next/static/image/public/${filename}.jpg`,
```

Comment:
How was this testing working before?

It seems like we need a new test to capture the issue described in #29001 .

Otherwise I would have expected the status to be something other than 200.

Reply:
@styfle From my understanding, that test was passing before because filenames with multiple dots didn't work, as the code considered the file would have a single dot only (the file extension). So if you had `file.name.with.dots.png` it'd return only `file.png`. 

With these changes, that assertion must expect `test.480a01....webp`, not only `test.webp`, because multiple dots are supported now and the `query.url` uses`test.480a01....jpg`.

I'm not sure if we need more updates to the integration specs. It seems it tests that when sending a request with a filename in query params, its response has the same name and the optimized extension on the 'Content-Disposition' response header.

I've added some unit specs covering the case [here](https://github.com/vercel/next.js/blob/2b0ab38c9d423948bbfe19c940ac13af6005ee1f/test/unit/image-optimizer/get-file-name-with-extension.test.ts).

Reply:
@matheusazzi The PR description says

> Next.js tries to optimize the image but fails to load it in the browser when the image file name has dots.

That sounds like its failing to render, not setting the wrong `Content-Disposition` header.

Why was this test not failing?
And can you add an integration test for an image that is failing?